### DISCUSSION TITLE
Embed console or show console output in the GUI

### DISCUSSION THREAD

**Discussion by Beelzebub2** [TomSchimansky/CustomTkinter#623] [Ideas]
is it possible to have a small console embedded in the GUI?

**Comment by abhaymakes** [comment]
@Beelzebub-uwu you might be looking for something like [this](https://github.com/Saadmairaj/tkterminal).
However, this is not the best thing, you could consider making a console from yourself.

**Reply by Beelzebub2** [reply]
That's what i needed thanks

**Reply by Beelzebub2** [reply]
I don't really have the skills to make it from zero

**Comment by abhaymakes** [comment]
@Beelzebub-uwu Consider marking this as answer if it helped. :)

**Reply by Beelzebub2** [reply]
How?😅

**Reply by abhaymakes** [reply]
There might be a checkmark thing on my comment, click it and it's done.

**Reply by Akascape** [reply]
@TheFallen-Cat Galaxy Brain, but this is not a QnA discussion... LOL

**Reply by abhaymakes** [reply]
@Akascape Damn xD

**Reply by Beelzebub2** [reply]
> @Akascape Damn xD

I did upvote it tho :)

**Comment by Akascape** [comment]
@Beelzebub-uwu **Here is the example you need:**

![program](https://user-images.githubusercontent.com/89206401/202501186-d4d3cbb0-fe92-4e45-bf71-1aa5b919e4bd.jpg)

## Code
```python
import customtkinter
from tkterminal import Terminal

def run_command():
    terminal.run_command("python --version")

root = customtkinter.CTk()
root.geometry("400x250")

terminal = Terminal(root, height=10, background="black", foreground="white",
                    relief="flat", width=45, pady=5, padx=5)

terminal.grid(padx=12, pady=10)

button = customtkinter.CTkButton(root, text="Check Python version", command=run_command).grid()

root.mainloop()
```
Please refer the documentation of [tkterminal](https://github.com/Saadmairaj/tkterminal)

**Reply by Beelzebub2** [reply]
I haven't started working on it but my objective was to show the output of the script that's running on the embedded terminal

**Comment by rafugafu** [comment]
[PyTkTerm](https://github.com/rafugafu/pytkterm) implements a full terminal inside a tkinter `Text` widget widget.
